cmd/ctr/commands/run/run.go

```diff
@@ -18,19 +20,128 @@ import (
 	"github.com/urfave/cli"
 )
 
-func withEnv(context *cli.Context) containerd.SpecOpts {
+type containerConfig struct {
+	Checkpoint  string   `toml:"checkpoint"`
+	Cwd         string   `toml:"cwd"`
+	Env         []string `toml:"env"`
+	Label       []string `toml:"label"`
+	Mount       []string `toml:"mount"`
+	NetHost     bool     `toml:"net_host"`
+	ReadOnly    bool     `toml:"read_only"`
+	Runtime     string   `toml:"runtime"`
+	Snapshotter string   `toml:"snapshotter"`
+	TTY         bool     `toml:"tty"`
+	// Unix-specific field
+	RootFS bool `toml:"rootfs"`
+	// Windows-specific field
+	Layer []string `toml:"layers"`
+}
+
+// ContainerFlags are cli flags specifying container options
+var ContainerFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "config,c",
+		Usage: "path to the configuration file",
+	},
+	cli.StringFlag{
+		Name:  "checkpoint",
+		Usage: "provide the checkpoint digest to restore the container",
+	},
+	cli.StringFlag{
+		Name:  "cwd",
+		Usage: "specify the working directory of the process",
+	},
+	cli.StringSliceFlag{
+		Name:  "env",
+		Usage: "specify additional container environment variables (i.e. FOO=bar)",
+	},
+	cli.StringSliceFlag{
+		Name:  "label",
+		Usage: "specify additional labels (i.e. foo=bar)",
+	},
+	cli.StringSliceFlag{
+		Name:  "mount",
+		Usage: "specify additional container mount (ex: type=bind,src=/tmp,dest=/host,options=rbind:ro)",
+	},
+	cli.BoolFlag{
+		Name:  "net-host",
+		Usage: "enable host networking for the container",
+	},
+	cli.BoolFlag{
+		Name:  "read-only",
+		Usage: "set the containers filesystem as readonly",
+	},
+	cli.StringFlag{
+		Name:  "runtime",
+		Usage: "runtime name (io.containerd.runtime.v1.linux, io.containerd.runtime.v1.windows, io.containerd.runtime.v1.com.vmware.linux)",
+		Value: fmt.Sprintf("io.containerd.runtime.v1.%s", runtime.GOOS),
```

Comment:
see here @dnephin

Reply:
Oo, we need to figure this one out. `GOOS` isn't the correct value here.